R/utils.R

```diff
@@ -58,39 +58,59 @@ py_global_env <- function() {
   reticulate::py
 }
 
-# Create a duplicate of a Python environment
-#
-# @examples
-# reticulate::py_run_string("x = 3")
-# new_py_envir <- duplicate_py_env(py)
-# new_py_envir$items()
-#
-# @param module the `py` module accessed via `reticulate`
-#
-# @return a Python `Dict` or dictionary that is not converted to an R data type
-# @keywords internal
-duplicate_py_env <- function(module) {
+#' Create a duplicate of a Python environment
+#'
+#' @return a Python `Dict` or dictionary
+#'
+#' @noRd
+#' @keywords internal
+#'
+#' @examples
+#' \dontrun{
+#' reticulate::py_run_string("x = 3")
+#' new_py_envir <- duplicate_py_env()
+#' new_py_envir$items()
+#' }
+duplicate_py_env <- function() {
```

Comment:
If the overall approach is working for you, lets go ahead and update this function name to follow a `py_` prefix convention. And since it's now duplicating the global env, let's add that in there. Also the action is more _copy_ than _duplicate_.

```suggestion
py_copy_global_env <- function() {
```